sdks/python/apache_beam/ml/inference/sklearn_inference_test.py

```diff
@@ -338,6 +372,42 @@ def test_pipeline_pandas(self):
       assert_that(
           actual, equal_to(expected, equals_fn=_compare_dataframe_predictions))
 
+  def test_pipeline_pandas_custom_batching(self):
+    temp_file_name = self.tmpdir + os.sep + 'pickled_file'
+    with open(temp_file_name, 'wb') as file:
+      pickle.dump(build_pandas_pipeline(), file)
+
+    def batch_validator_pandas_inference_fn(
+        model: BaseEstimator,
+        batch: Sequence[numpy.ndarray],
+        inference_args: Optional[Dict[str, Any]] = None) -> Any:
+      if len(batch) != 5:
+        raise Exception(
+            'Expected batch of size 5, received batch of size {}'.format(
```

Comment:
(personal opinion) I find f-strings easier to read, e.g. `f'Expected batch of size 5, received batch of size {len(batch)}'`.

Reply:
I have no preference here - updated